src/Symfony/Component/Notifier/Bridge/LineBot/LineBotTransportFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\LineBot;
+
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+
+/**
+ * @author Yi-Jyun Pan <me@pan93.com>
+ */
+final class LineBotTransportFactory extends AbstractTransportFactory
+{
+    private const SCHEME = 'linebot';
+
+    protected function getSupportedSchemes(): array
+    {
+        return [self::SCHEME];
+    }
+
+    public function create(Dsn $dsn): LineBotTransport
+    {
+        if (self::SCHEME !== $dsn->getScheme()) {
+            throw new UnsupportedSchemeException($dsn, self::SCHEME, $this->getSupportedSchemes());
+        }
+
+        $accessToken = $dsn->getUser();
+        if (null === $accessToken) {
+            throw new \InvalidArgumentException('You must provide a channel access token as "user" in the DSN.');
+        }
+
+        $receiver = $dsn->getRequiredOption('receiver');
+        if (!\is_string($receiver)) {
+            throw new \InvalidArgumentException('"receiver" must be a string user ID.');
+        }
```

Comment:
```suggestion
```

Reply:
![CleanShot 2024-10-11 at 07 53 06@2x](https://github.com/user-attachments/assets/5f1bf002-cfa6-4b70-8c4c-16a4f740bd71)

It returns a `mixed` which makes PHPStan sad. Maybe I can change it to a `assert`?